ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MachineAccessIntegrationForm.tsx

```diff
@@ -0,0 +1,377 @@
+                                                            }
+                                                        >
+                                                            <SelectSingle
+                                                                id={`mappings[${index}].role`}
+                                                                value={
+                                                                    values.mappings[`${index}`].role
+                                                                }
+                                                                isDisabled={!isEditable}
+                                                                handleSelect={setFieldValue}
+                                                                direction="up"
+                                                                placeholderText="Select a role"
+                                                            >
+                                                                {roles.map(({ name }) => (
+                                                                    <SelectOption
+                                                                        key={name}
+                                                                        value={name}
+                                                                    />
+                                                                ))}
+                                                            </SelectSingle>
+                                                        </FormGroup>
+                                                    </FlexItem>
+                                                    {isEditable && (
+                                                        <FlexItem>
+                                                            <Button
+                                                                variant="plain"
+                                                                aria-label="Delete rule"
+                                                                style={{
+                                                                    transform: 'translate(0, 42px)',
+                                                                }}
+                                                                onClick={() =>
+                                                                    arrayHelpers.remove(index)
+                                                                }
+                                                            >
+                                                                <TrashIcon />
+                                                            </Button>
+                                                        </FlexItem>
+                                                    )}
+                                                </Flex>
+                                            ))}
```

Comment:
A note in the mock specifies that there should always be at least one rule line, but I don't see where we are enforcing that requirement in the code.

https://docs.google.com/presentation/d/1GgxiqdjI2dLfexmdheMeqFe_8vHE28iqbcbJchnwvc0/edit#slide=id.g2a8b12867b7_0_1952

Is the mock incorrect?

Reply:
Based on backend implementation, there can be 0 rules, even though without a default role I'm not sure if such configuration makes much sense. @dhaus67 WDYT?

Reply:
Yeah we can enforce it on the UI to specify at least 1 rule and also add within the validation that _at least_ one rule has to be given.